### ISSUE TITLE
about a problem

### ISSUE THREAD

**Issue by rhyhck** [QingtaoPan/DuSSS#1]
Hello, author, is this sentence wrong? Isn't calculation of loss based on similarity, but the code is written in cos_ori:

https://github.com/QingtaoPan/DuSSS/blob/351b6d0d760d1510c71965452737370f48fbaf99/code/nets/text_mask/novel/objectives.py#L44

**Comment by rhyhck** [comment]
I have another question: since the average loss of i2t and t2i is calculated in this line, why is t2i calculated separately in this line?
   loss_it = (F.cross_entropy(cos_ori, labels) + F.cross_entropy(cos_ori.transpose(0, 1), labels)) / 2
    
   loss_t2i = objectives.compute_contrast_t2i(text_emb=text_feat, img_emb_aug=image_feat_m, ret=ret, temp=self.temp)

**Comment by rhyhck** [comment]
I have another question: since the average loss of i2t and t2i is calculated in this line, why is t2i calculated separately in this line?

https://github.com/QingtaoPan/DuSSS/blob/351b6d0d760d1510c71965452737370f48fbaf99/code/nets/text_mask/novel/objectives.py#L44

https://github.com/QingtaoPan/DuSSS/blob/351b6d0d760d1510c71965452737370f48fbaf99/code/vlm_pretrain/model_novel.py#L213

**Comment by rhyhck** [comment]
Hi, I have another question. What's the difference between these two sentences? Why does a second call to self.con_img_gau_encoder count as an enhanced feature?
https://github.com/QingtaoPan/DuSSS/blob/351b6d0d760d1510c71965452737370f48fbaf99/code/vlm_pretrain/model_novel.py#L196

https://github.com/QingtaoPan/DuSSS/blob/351b6d0d760d1510c71965452737370f48fbaf99/code/vlm_pretrain/model_novel.py#L198

Look forward to your response~ Thank you

**Comment by QingtaoPan** [comment]
(1) Thank you for pointing out the issue. I made a mistake here, I should replace cso'ori with similarity.
(2) In loss_i2t = objectives.compute_contrast_i2t(img_emb=image_feat, text_emb_aug=text_feat_m, ret=ret, temp=self.temp) and loss_t2i = objectives.compute_contrast_t2i(text_emb=text_feat, img_emb_aug=image_feat_m, ret=ret, temp=self.temp), the image_feat and text_feat_m are different from text_feat and image_feat_m, so the loss_t2i is different from loss_it.
(3) Thank you for pointing out the mistake. In  img_mu_aug, img_logsigma_aug, _ = self.con_img_gau_encoder(image_embeds_b_197_512, mask=None), image_embeds_b_197_512 should be image_embeds_b_197_512_m.
